### PR TITLE
Refactored `kestest` integration tests

### PR DESCRIPTION
**Pull request by shtripat**
This PR adds a cleanup method to make sure each test works in its entirety and there are no left outs.
Also all tests use the same common functions for scenarios like `create`, `import`, `generate`, `encrypt` and `decrypt` etc.

Different kind of tests can be executed using `go test -v -run="<test>"` command.
e.g. `go test -v -run="TestGatewayMem"` or `go test -v -fs.path=./keys -run="TestGatewayFS"`

### CODE REVIEW

File: kestest/gateway_test.go [hunk 1]
```diff
@@ -633,3 +626,28 @@ func mustDecodeB64(s string) []byte {
 	}
 	return b
 }
+
+func clean(ctx context.Context, client *kes.Client, pattern string, t *testing.T) {
```
Comment (aead) [inline kestest/gateway_test.go:630]: You can omit the pattern here and use the match all pattern `*` in `ListKeys`.

**Review by aead — CHANGES_REQUESTED** [review]
Minor tweaks. Also consider a descriptive commit message as suggested by the contributing doc.
Ref: https://github.com/minio/kes/blob/master/CONTRIBUTING.md#commit-message

File: kestest/gateway_test.go (hunk 1 above)
Comment (shtripat) [reply kestest/gateway_test.go:630]: > You can omit the pattern here and use the match all pattern `*` in `ListKeys`.

Ok, makes sense. Will do the changes.

File: kestest/gateway_test.go [hunk 2]
```diff
@@ -633,3 +626,28 @@ func mustDecodeB64(s string) []byte {
...
+
+func clean(ctx context.Context, client *kes.Client, t *testing.T) {
+	iter, err := client.ListKeys(ctx, "*")
+	if err != nil {
+		t.Fatalf("Cleanup: failed to list keys: %v", err)
+	}
+	defer iter.Close()
+
+	names := []string{}
+	keysInfo, err := iter.Values(-1)
```
Comment (aead) [inline kestest/gateway_test.go:638]: `iter.Values(-1)` already collects all `KeyInfo` objects into one slice. You don't need to clone this slice again. 
You can simply remove:
```
for _, keyInfo := range keysInfo {
	names = append(names, keyInfo.Name)
}
```

An just do one for loop:
```
for _, info := range keyInfos {
		if err = client.DeleteKey(ctx, info.Name); err != nil {
			t.Errorf("Cleanup: failed to delete '%s': %v", name, err)
		}
	}
```

Comment (shtripat) [reply kestest/gateway_test.go:637]: Done

**Review by aead — APPROVED** [review]

**Pull request merged**